### ISSUE TITLE
Refactoring a meaningful system type mailbox session

### ISSUE THREAD

**Issue by quantranhong1999** [linagora/james-project#4586]
### WHY

 - Current MailboxSession code brings limitations on what we can do (good luck to implement curl -XDELETE /msgs/:messageId)
 - We are not able to distinguish (eg in logs) between user actions and stuff triggered via a webadmin task. Only using users brings in traceability issues...

### HOW
- Refactor `createSystemSession` -> `createUserSession` (quickly change system -> user session in code)
- Create `createSystemSession` which creates a session with a meaningful System type
- Adding tests to make sure System type session have the full right access to arbitrary user resources (rename mbx, delete mbx, create mbx, copy msg, setFlags, delete msg, move msg, read msg, search mbx, search msg)

#### Concern
- Scope of `TypeSession` refactoring:
  - Option 1: 2 types of session (User/System): easier to archive our needs.
  - Option 2: 4 types of session proposed in the ML (User/ImpersonatedUser/DelegatedUser/System): more comprehensive and good for audit actions (by who) but come with bigger refactoring effort (action's actor information refactoring).
- `createSystemSession` with username is meaningless -> use a static system user?

#### Related stuff (raised by Rene long ago)
- Rational: https://github.com/linagora/james-project/pull/3035#discussion_r363684700
TLDR: should never `mailbox-store` (low-level code) outside mailbox component
-> To fully archive this would need much refactoring in James, refactoring system type MBS could be one of easy ones.
- Mailing list discussion: https://lists.apache.org/thread/chmylvglv3mqnmrgwyg49qwmbc5nxfff
- ADR (Mailbox API visibility and usage): https://github.com/linagora/james-project/pull/3333
- PR system MBS: https://github.com/linagora/james-project/pull/3038

**Comment by chibenwa** [comment]
I like option 1, we could then migrate to option 2 later on...

> createSystemSession with username is meaningless -> use a static system user?

No but we are interested by the component eg the task...

IMO this deserves to be discussed on the ML rather than on a Linagora repo.

**Comment by chibenwa** [comment]
Should we close @quantranhong1999 ?

**Comment by quantranhong1999** [comment]
It is still a thing we want?

**Comment by chibenwa** [comment]
No at least not now...
